apps/web/src/features/spaces/components/SpaceAddressBook/EditContactDialog.tsx

```diff
@@ -0,0 +1,189 @@
+import { trackEvent } from '@/services/analytics'
+import { SPACE_EVENTS } from '@/services/analytics/events/spaces'
+import type { SpaceAddressBookEntry } from '../../types'
+import useChains from '@/hooks/useChains'
+import type { ContactField } from './AddContact'
+
+type EditContactDialogProps = {
+  entry: SpaceAddressBookEntry
+  onClose: () => void
+}
+
+// TODO: Replace with actual CGW API calls
+const createAddressBookEntry = async (address: string, name: string, chainId: string) => {
+  console.log('CREATE:', { address, name, chainId })
+  return
+}
+
+const updateAddressBookEntry = async (id: string, name: string) => {
+  console.log('UPDATE:', { id, name })
+  return
+}
+
+const deleteAddressBookEntry = async (id: string, chainId: string) => {
+  console.log('DELETE:', { id, chainId })
+  return
+}
+
+const EditContactDialog = ({ entry, onClose }: EditContactDialogProps) => {
+  const [error, setError] = useState('')
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const { configs } = useChains()
+
+  const defaultNetworks = entry.networks
+    .map((network) => {
+      return configs.find((chain) => chain.chainId === network.chainId) as ChainInfo
+    })
+    .filter(Boolean)
+
+  const defaultValues = {
+    name: entry.name,
+    address: entry.address,
+    networks: defaultNetworks,
+  }
+
+  const methods = useForm<ContactField>({
+    mode: 'onChange',
+    defaultValues,
+  })
+
+  const { handleSubmit, formState, control, reset, watch } = methods
+
+  const { errors } = formState
+
+  // Watch for changes in name and networks
+  const watchedName = watch('name')
+  const watchedNetworks = watch('networks')
+
+  // Check if any changes were made
+  const hasChanges = useMemo(() => {
+    const nameChanged = watchedName !== entry.name
+
+    const originalChainIds = entry.networks.map((network) => network.chainId).sort()
+    const currentChainIds = watchedNetworks.map((network) => network.chainId).sort()
+    const networksChanged =
+      originalChainIds.length !== currentChainIds.length ||
+      originalChainIds.some((id, index) => id !== currentChainIds[index])
+
+    return nameChanged || networksChanged
+  }, [watchedName, watchedNetworks, entry.name, entry.networks])
+
+  const handleClose = () => {
+    reset(defaultValues)
+    setError('')
+    onClose()
+  }
+
+  const onSubmit = handleSubmit(async (data) => {
+    try {
+      setIsSubmitting(true)
+      trackEvent({ ...SPACE_EVENTS.ADD_ADDRESS_SUBMIT })
+
+      const selectedChainIds = data.networks.map((network) => network.chainId)
+      const originalChainIds = entry.networks.map((network) => network.chainId)
+
+      // Determine which networks need to be created, updated, or deleted
+      const toCreate = selectedChainIds.filter((chainId) => !originalChainIds.includes(chainId))
+      const toDelete = originalChainIds.filter((chainId) => !selectedChainIds.includes(chainId))
```

Comment:
It looks like this will be handled by the CGW where we will have one upsert endpoint that we send all the selected chainIds to and it will then decide what to delete/create/edit.